src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -290,56 +290,44 @@ public function testGetErrorOutput()
         $this->assertEquals(3, preg_match_all('/ERROR/', $p->getErrorOutput(), $matches));
     }
 
-    public function testGetIncrementalErrorOutput()
+    /**
+     * @dataProvider provideIncrementalOutput
+     */
+    public function testIncrementalOutput($getOutput, $getIncrementalOutput, $uri)
     {
-        // use a lock file to toggle between writing ("W") and reading ("R") the
-        // error stream
-        $lock = tempnam(sys_get_temp_dir(), get_class($this).'Lock');
-        file_put_contents($lock, 'W');
+        $lock = tempnam(sys_get_temp_dir(), __FUNCTION__);
+
+        $p = $this->getProcess(sprintf('%s -r %s', self::$phpBin, escapeshellarg('file_put_contents($s = \''.$uri.'\', $s); flock(fopen('.var_export($lock, true).', \'r\'), LOCK_EX); file_put_contents($s, \'foo\');')));
```

Comment:
Why using the uri as content to be written ? this makes the test harder to read IMO. Couldn't we use some hardcoded content instead, and use the uri only as uri ?